src/ai/backend/manager/registry.py

```diff
@@ -237,6 +237,7 @@ class AgentRegistry:
     pending_waits: set[asyncio.Task[None]]
     database_ptask_group: aiotools.PersistentTaskGroup
     webhook_ptask_group: aiotools.PersistentTaskGroup
+    agent_installed_images: dict[AgentId, set[str]]
```

Comment:
we are caching image info to redis, should we cache locally again?

Reply:
Yes, this is necessary to detect images that need to be removed.